src/subject_name/dns_name.rs

```diff
@@ -584,6 +698,10 @@ fn is_valid_dns_id(
     true
 }
 
+fn is_valid_wildcard_dns_id(hostname: untrusted::Input) -> bool {
```

Comment:
This appears to be single-use... I'd probably inline it at the callsite.

Reply:
I went ahead and inlined this and its sibling `is_valid_reference_dns_id`. 

In general I think there's an existing pattern of making these smaller single-use helpers on top of more generic functions (e.g. `presented_id_matches_reference_id` and `presented_id_matches_constraint`). It might make sense to consider whether we want to inline those as a general policy. I've left those two as-is for now since they're not touched by the work in this branch.